python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_basic_structured_config.py

```diff
@@ -349,6 +359,16 @@ def pipeline_requires_config():
         "loggers": {},
     }
 
+    result_with_runconfig = validate_run_config(
+        pipeline_requires_config, RunConfig(ops={"requires_config": {"foo": "bar"}})
+    )
```

Comment:
hmmm seems like we should have RunConfig take only the typed variant? If we do that i seems like we could have a better type signature

Reply:
Having `ops` typed as `Dict[str, Config]` would be a bit nicer, but it does then gate this API to people who have fully cut over to structured config. One of the cases that would be nice to support is gradual migration, e.g. defining new ops/assets using the new syntax.